### PR TITLE
Chapter1/homework/5

### PR DESCRIPTION
**Pull request by ShadowEnder**
Was able to get all tests to pass except for one for InsertLast.

Seems that whatever value you put in just winds up as zero for some reason.

### CODE REVIEW

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter1/Program.cs  1
- Src/BootCamp.Chapter1/ArrayOperations.cs  10
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4954424&bid=16373582)

File: Src/BootCamp.Chapter1/Program.cs [hunk 1]
```diff
@@ -1,9 +1,19 @@
...
+namespace BootCamp.Chapter1
 {
     class Program
     {
         static void Main(string[] args)
         {
+            var arr = new int[3];
+            arr[0] = 3;
+            arr[1] = 1;
+            //arr[2] = 1;
```
Comment (Almantask) [inline Src/BootCamp.Chapter1/Program.cs]: Codacy Issue found: [Remove this commented out code.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4954424)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter1/Program.cs  1
- Src/BootCamp.Chapter1/ArrayOperations.cs  10
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4954424&bid=16373582)

File: Src/BootCamp.Chapter1/ArrayOperations.cs [hunk 2]
```diff
@@ -9,7 +9,27 @@ public static class ArrayOperations
         /// <param name="array">Input array in a random order.</param>
         public static void Sort(int[] array)
         {
-            // ToDo: implement.
+            if (array == null || array.Length == 0)
```
Comment (Almantask) [inline Src/BootCamp.Chapter1/ArrayOperations.cs]: Nice to return early :)

File: Src/BootCamp.Chapter1/ArrayOperations.cs [hunk 3]
```diff
@@ -9,7 +9,27 @@ public static class ArrayOperations
...
+            {
+                return;
+            }
+
+            int temp;
+            // traverse = to array length
+            for (int i = 0; i < array.Length - 1; i++)
+            {
+                // traverse i + 1 to array length
```
Comment (Almantask) [inline Src/BootCamp.Chapter1/ArrayOperations.cs]: We don't need redundant comments, code speaks for itself

Comment (cptcrunchy) [reply Src/BootCamp.Chapter1/ArrayOperations.cs]: One of your conditions is incorrect inside of the InsertAt method.

you have 
if (array.Length == 0 && index > 0 || **index > array.Length - 1** || index < 0) ...

but this should be
if (array.Length == 0 && index > 0 || **index < array.Length - 1** || index < 0) ...

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter1/Program.cs  1
- Src/BootCamp.Chapter1/ArrayOperations.cs  10
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4954424&bid=16373582)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter1/Program.cs  1
- Src/BootCamp.Chapter1/ArrayOperations.cs  9
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4954424&bid=16373582)

**Comment by ShadowEnder** [comment]
Just want to note that even though it has passed on tests I don't think it is actually working.
Example:
In program.cs I have:
`var arr = new int[3];
            arr[0] = 3;
            arr[1] = 1;
            ArrayOperations.InsertLast(arr, 222);
            foreach(var i in arr)
            {
                Console.WriteLine(i);
            }`
which returns:
3
1
0
I get the same result if I use InsertFirst or InsertAt (no matter the index) instead of InsertLast (same for using different numbers to insert).

**Pull request closed without merging**